From a5f418b919b5463ae7d5acb109d0026dc0b0a522 Mon Sep 17 00:00:00 2001
From: Yongqiang Sun <yongqiang.sun@amd.com>
Date: Wed, 23 Nov 2016 16:21:39 -0500
Subject: [PATCH 1720/1722] drm/amd/amdgpu: Add mutex to amdgpu connector.

	Hot plug some model of DP monitor cause gpio conflict due to
	get a short pulse interrupt when reading edid data. By adding
	mutex, it will protect hpd interrupt handle to avoid that
	conflict when short pulse irq happened.

Reviewed-by: Harry Wentland <harry.wentland@amd.com>
Change-Id: I914714656aae06504632051b34352cd7b7c6df4c
Signed-off-by: Yongqiang Sun <yongqiang.sun@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h b/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h
index 965bb7a..dba0f00 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_mode.h
@@ -599,7 +599,7 @@ struct amdgpu_connector {
 	/*freesync caps*/
 	struct mod_freesync_caps caps;
 
-
+	struct mutex hpd_lock;
 
 };
 
-- 
2.7.4

